### PR TITLE
Document msbuild telemetry datapoints

### LINKED ISSUE #10799: Update external documentation on telemetry
### Context

Followup of https://github.com/dotnet/msbuild/issues/10634. Specifically this step:

- [x] Document the datapoints in our wiki
- [ ] Update SDK telemetry docs in [dotnet/docs](https://github.com/dotnet/docs/issues/new?template=z-customer-feedback.yml&pageUrl=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fdotnet%2Fcore%2Ftools%2Ftelemetry&pageQueryParams&contentSourceUrl=https%3A%2F%2Fgithub.com%2Fdotnet%2Fdocs%2Fblob%2Fmain%2Fdocs%2Fcore%2Ftools%2Ftelemetry.md&documentVersionIndependentId=50433ae5-4687-fe5f-75ab-a47e1a706777&feedback=%0A%0A%5BEnter+feedback+here%5D%0A&author=%40KathleenDollard&metadata=*+ID%3A+c136055b-bb04-17b6-19b2-42d4610c1a70+%0A*+Service%3A+**dotnet-fundamentals**) - https://github.com/dotnet/docs/issues/43034

As we need to be transparent with our users about telemetry we collect

### PR DESCRIPTION
**Pull request by JanKrivanek**
Fixes #10799

### Context

Document telemetry collected by msbuild to be transparent with our users about the data we collect.
This is documenting datapoints added in NET9, but also filling gap of missing documentation of pre-existing telemetry datapoints

### Notes

Documentation only, no code changes

### CODE REVIEW

File: documentation/wiki/CollectedTelemetry.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+# MSBuild Telemetry
+
+MSBuild emits and collects telemetry to guide decisions on modernization and optimization investments. Currently the MSBuild collects telemetry only when run from SDK host (mostly the `dotnet build` and `dotnet msbuild` commands). For more details please refer to [the official SDK telemetry documentation](https://learn.microsoft.com/en-us/dotnet/core/tools/telemetry).
+
+Visual Studio collects some build related telemetry (TODO: link the specific info here) - but that is not leveraging any MSBuild instrumentation, but rather information about count and duration of MSBuild API invocations from the caller point of view.
```
Comment (YuliiaKovalova) [inline documentation/wiki/CollectedTelemetry.md:5]: TODO ? :)

**Review by YuliiaKovalova — APPROVED** [review]

File: documentation/wiki/CollectedTelemetry.md (hunk 1 above)
Comment (JanKrivanek) [reply documentation/wiki/CollectedTelemetry.md]: Unfortunately I couldn't locate any official document that would list in detail what datapoints are being collected for VS. Hence the TODO to tri to dig for it.

I still wasn't successful - so I rephrased and mentioned official VS telemetry compliance doc

File: documentation/wiki/CollectedTelemetry.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+### BuildCheck
+
+Expressed and collected via [BuildCheckTelemetry type](https://github.com/dotnet/msbuild/blob/94941d9cb26bb86045452b4a174a357b65a30c99/src/Framework/Telemetry/BuildCheckTelemetry.cs)
+
+#### BuildCheck Run
+
+| SDK versions | Data |
+|--------------|------|
+| >= 9.0.1     | Corelation guid for the run |
```
Comment (baronfel) [inline documentation/wiki/CollectedTelemetry.md:46]: Same here - 9.0.1xx, not 9.0.1

**Review by baronfel — APPROVED** [review]
This is a good summary - can you please call-out the use of the MSBuild Engine's telemetry capability? Meaning if a Task calls `.LogTelemetry`, that telemetry will be processed by our telemetry subsystems - the .NET SDK will not emit that telemetry (since it has an allowlist for telemetry data from MSBuild), but other MSBuild Hosts _may_ emit that telemetry.

File: documentation/wiki/CollectedTelemetry.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+| >= 9.0.1     | Count of enabled rules for the run |
+| >= 9.0.1     | Count of enabled custom rules for the run |
+| >= 9.0.1     | Count of violations encountered for the run |
+| >= 9.0.1     | Execution time spent by BuildCheck infrastructure and rules |
+
+#### BuildCheck Rule in a run
+
+| SDK versions | Data |
+|--------------|------|
+| >= 9.0.1     | Corelation guid for the run. |
```
Comment (maridematte) [inline documentation/wiki/CollectedTelemetry.md:56]: Do we keep track of runs using a guid for each run?

File: documentation/wiki/CollectedTelemetry.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+# MSBuild Telemetry
+
+MSBuild emits and collects telemetry to guide decisions on modernization and optimization investments. Currently the MSBuild collects telemetry only when run from SDK host (mostly the `dotnet build` and `dotnet msbuild` commands). For more details please refer to [the official SDK telemetry documentation](https://learn.microsoft.com/en-us/dotnet/core/tools/telemetry).
```
Comment (maridematte) [inline documentation/wiki/CollectedTelemetry.md:3]: We should default links to no region: https://learn.microsoft.com/dotnet/core/tools/telemetry

File: documentation/wiki/CollectedTelemetry.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+Visual Studio collects some build related telemetry - but that is not leveraging any MSBuild instrumentation, but rather information about count and duration of MSBuild API invocations from the caller point of view. For general information about telemetry being collected by Visual Studio Family of products and regulations compliance please refer to [the official documentation](https://learn.microsoft.com/en-us/compliance/regulatory/gdpr-dsr-visual-studio-family).
```
Comment (maridematte) [inline documentation/wiki/CollectedTelemetry.md:5]: Another link with region: https://learn.microsoft.com/compliance/regulatory/gdpr-dsr-visual-studio-family

File: documentation/wiki/CollectedTelemetry.md (hunk 3 above)
Comment (JanKrivanek) [reply documentation/wiki/CollectedTelemetry.md:56]: Yes - just for the buildchecks though.

A guid is generated after build run and added to each buildcheck datapoint for that run - to be able to connect those together

**Review by maridematte — APPROVED** [review]

**Pull request merged**